### PR TITLE
API Routes - Slide

### PR DESCRIPTION
**Pull request by tawnyzhao**
## Notion ticket link

[API Routes - Slide](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=106d6eaaae5c4c46bd9e349d25025442)

## Implementation description
* Add create, update, delete, read routes for slide
* Create `POST module/<id>/slide`
* Update `PATCH slide/<id>`
* Delete `DELETE module/<id>/slide/<sid>`
* Read `GET slide/<id>`
* Remove unused properties of slide (prev and next)
* Add Types for error responses
* Fix types on Module model and module interface

## Steps to test
1. Make a module
2. Add a slide to the module
3. Update the slide
4. Read the slide for the update to persist
5. Delete the slide from the module

## What should reviewers focus on?
* Setup for the routes are correct

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/slide/[id].ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import connectDB from "../utils/mongoose";
+import { del } from "./delete";
+import { get } from "./get";
+import { patch } from "./patch";
+
+const index = async (
+    req: NextApiRequest,
+    res: NextApiResponse<ErrorResponse>,
+) => {
+    switch (req.method) {
```
Comment (nimwijetunga) [inline pages/api/slide/[id].ts:12]: What's the difference between `/module/id/slide/id` and `/slide/id` routes (can we have slides that are not associated with a module)?

**Review by nimwijetunga — APPROVED** [review]
lgtm!

File: pages/api/slide/delete.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { NextApiRequest, NextApiResponse } from "next";
+import { ErrorResponse } from "types/ErrorResponse";
+import { Module } from "../models/module";
+import { Slide } from "../models/slide";
+
+const del = async (
+    req: NextApiRequest,
+    res: NextApiResponse<ErrorResponse>,
+): Promise<void> => {
+    const { id, sid } = req.query;
```
Comment (mkanoria) [inline pages/api/slide/delete.ts:11]: Do we need both slide ID and module ID here? Maybe we add a Module ID inside each slide instead?

**Comment by ChinemeremChigbo** [comment]
Creating and deleting slides works as intended and the code looks good. Well done!

**Review by ChinemeremChigbo — APPROVED** [review]

**Comment by mkanoria** [comment]
@tawnyzhao feel free to merge this in once the conflicts are resolved

**Pull request merged**